crates/core/src/client/mod.rs

```diff
@@ -0,0 +1,324 @@
+mod config;
+pub(crate) mod inner;
+
+#[cfg(test)]
+mod tests;
+
+use std::{
+    collections::HashMap,
+    sync::{Arc, Mutex},
+};
+
+use config::*;
+use futures::future::try_join_all;
+use inner::LiveViewClientInner;
+use phoenix_channels_client::{Event, Payload, SocketStatus, JSON};
+use reqwest::header::CONTENT_TYPE;
+
+use crate::{
+    callbacks::*,
+    dom::ffi::{self},
+    error::LiveSocketError,
+    live_socket::{navigation::NavOptions, ConnectOpts, LiveChannel, LiveFile, Method},
+};
+
+#[derive(uniffi::Object, Default)]
+/// A configuration interface for building a [LiveViewClient].
+/// Options on this object will used for all http and websocket connections
+/// through out the current session.
+///
+/// Additionally provides the [LiveViewClient] with callabcks and essential functionality,
+/// without proper configuration your client may not function properly.
+/// See [LiveViewClientBuilder::set_persistence_provider]
+pub struct LiveViewClientBuilder {
```

Comment:
I will admit that I don't truly know is docstrings should go between the derive and the struct it's modifying but if it works both ways, I suggest putting the docstring before the derive. There's just too much space (7 lines) between the derives and the struct that's being derived.